asm/inst_other.go

```diff
@@ -71,6 +71,16 @@ func (fgen *funcGen) newSelectInst(ident ir.LocalIdent, old *ast.SelectInst) (*i
 	return &ir.InstSelect{LocalIdent: ident, Typ: typ}, nil
 }
 
+// newSelectInst returns a new IR select instruction (without body but with
+// type) based on the given AST select instruction.
+func (fgen *funcGen) newFreezeInst(ident ir.LocalIdent, old *ast.FreezeInst) (*ir.InstFreeze, error) {
+	x, err := fgen.irTypeValue(old.X())
```

Comment:
```suggestion
	typ, err := fgen.gen.irType(old.X().Typ())
```

Issue identified in https://github.com/llir/llvm/issues/132#issuecomment-613510792